app/controllers/proposal/collection_draft_proposals_controller.rb

```diff
@@ -2,7 +2,45 @@ module Proposal
   class CollectionDraftProposalsController < CollectionDraftsController
     skip_before_action :provider_set?
 
+    # TODO: Limit this to only the things a user is supposed to do, also need a new one for approver
+    # TODO: Also need one for functions which both can perform (e.g. show)
     before_action :ensure_non_nasa_draft_user
+    before_action(only: %I[submit rescind]) { set_resource }
```

Comment:
Is this the additive way of doing it, or is it just another way of using `before_action` and `only`?